src/inapp/utils.ts

```diff
@@ -232,6 +232,53 @@ const mediaQueryLg = global?.matchMedia?.(
 );
 const mediaQueryXl = global?.matchMedia?.('(min-width: 1301px)');
 
+/**
+ *
+ * @returns { HTMLIFrameElement } iframe with sandbox and hidden styling applied for of an inapp message to render with
+ */
+const generateSecuredIFrame = () => {
+  const iframe = document.createElement('iframe');
+  iframe.setAttribute('id', 'iterable-iframe');
+  // allow-popups and allow-top-navigation is to enable links for Safari since the iframe will block
+  // event handlers on elements in it preventing our custom link handling
+  iframe.setAttribute(
+    'sandbox',
+    'allow-same-origin allow-popups allow-top-navigation'
+  );
+  /*
+    _display: none_ would remove the ability to set event handlers on elements
+    so instead we choose to hide it visibly with CSS but not actually remove
+    its interact-ability
+
+    https://snook.ca/archives/html_and_css/hiding-content-for-accessibility
+  */
+  iframe.style.cssText = `
+    position: fixed !important;
+    top: 0;
+    left: 0;
+    height: 1px;
+    width: 1px;
+    overflow: hidden;
+  `;
+
+  return iframe;
+};
+
+/**
+ *
+ * @param html
+ * @returns { HTMLIFrameElement } iframe with html string of an inapp message wrapped in onload
+ */
+export const wrapWithIFrame = (html: string): HTMLIFrameElement => {
+  const iframe = generateSecuredIFrame();
+  iframe.onload = () => {
+    iframe.contentWindow?.document.write(
+      html.replaceAll(/\r?\n|\r/g, '').replaceAll('"', '\\"')
```

Comment:
You probably don't need to do this at all anymore. Calling write on (html) should work just fine. I suspect that the " to \" replacement will create some visual artifacts.

Reply:
Yep my bad missed commiting that change